packages/trpc/server/routers/viewer/slots/util.ts

```diff
@@ -378,73 +430,35 @@ export async function getAvailableSlots({ input, ctx }: GetScheduleOptions): Pro
   if (eventType.schedulingType === SchedulingType.ROUND_ROBIN && input.bookerEmail) {
-    let crmRoundRobinLeadSkip;
-    // See if CRM app is enabled and skip RR assignment
-    const eventTypeAppMetadata = eventType?.metadata?.apps;
-    for (const appKey in eventTypeAppMetadata) {
-      const app = eventTypeAppMetadata[appKey as keyof typeof eventTypeAppMetadata];
-      if (
-        app.enabled &&
-        typeof app.appCategories === "object" &&
-        app.appCategories.some((category: string) => category === "crm") &&
-        app.roundRobinLeadSkip
-      ) {
-        crmRoundRobinLeadSkip = app;
-        break;
-      }
-    }
+    const crmContactOwner = await getCRMContactOwnerForRRLeadSkip(
+      input.bookerEmail,
+      eventType?.metadata?.apps
+    );
+    const contactOwnerHost = hosts.find((host) => host.user.email === crmContactOwner);
 
-    if (crmRoundRobinLeadSkip) {
-      const crmCredential = await prisma.credential.findUnique({
-        where: {
-          id: crmRoundRobinLeadSkip.credentialId,
-        },
-        include: {
-          user: {
-            select: {
-              email: true,
-            },
-          },
-        },
-      });
-      if (crmCredential) {
-        const crm = new CrmManager(crmCredential);
-        const contact = await crm.getContacts(input.bookerEmail, true);
-        if (contact?.length) {
-          // Since this is enabled for round robin event, we iterate through hosts
-          const contactOwner = eventType.hosts.find((host) => host.user.email === contact[0].ownerEmail);
-          if (contactOwner) {
-            teamMember = contactOwner.user.email;
-            const contactOwnerIsRRHost = eventType.hosts.find(
-              (host) => host.user.email === teamMember && !host.isFixed
-            );
-            const otherHosts = contactOwnerIsRRHost
-              ? eventType.hosts
-                  .filter((host) => host.user.email !== contactOwner.user.email && host.isFixed)
-                  .map(({ isFixed, user }) => ({ isFixed, ...user }))
-              : eventType.hosts
-                  .filter((host) => host.user.email !== contactOwner.user.email)
-                  .map(({ isFixed, user }) => ({ isFixed, ...user }));
-
-            usersWithCredentials = [{ ...contactOwner.user, isFixed: true }, ...otherHosts];
-          }
-        }
-      }
-    }
-  }
+    if (contactOwnerHost) {
+      teamMember = contactOwnerHost.user.email;
+      const contactOwnerIsRRHost = !contactOwnerHost.isFixed;
 
-  if (!usersWithCredentials) {
-    if (eventType.schedulingType && !!eventType.hosts?.length) {
-      usersWithCredentials = eventType.hosts.map(({ isFixed, user }) => ({ isFixed, ...user }));
-    } else {
-      usersWithCredentials = eventType.users.map((user) => ({
-        isFixed: !eventType.schedulingType || eventType.schedulingType === SchedulingType.COLLECTIVE,
-        ...user,
-      }));
+      usersWithCredentials = usersWithCredentials.filter(
```

Comment:
remove contact owner, because it will be added later as a fixed host

if contact owner is a RR host remove all RR hosts